frontend/src/views/CalculateNutrients/CalculateNutrientsComponents/FertilizerModal.tsx

```diff
@@ -214,26 +344,152 @@ export default function FertilizerModal({
     }));
   };
 
-  const handleInputChanges = (changes: { [name: string]: string | number | undefined }) => {
-    const name = Object.keys(changes)[0];
-    const value = changes[name];
+  const handleInputChanges = useCallback(
+    (updates: { [key: string]: string | number | undefined }) => {
+      Object.entries(updates).forEach(([name, value]) => {
+        // eslint-disable-next-line prefer-const
+        let changes = structuredClone(updates);
 
-    if (name === 'fertilizerTypeId') {
-      setFilteredFertilizerOptions(
-        fertilizerOptions.filter(
-          (ele) =>
-            ele.dryliquid === fertilizerTypes.find((fertType) => fertType.id === value)?.dryliquid,
-        ),
-      );
-    }
+        if (['potassium', 'phosphorous', 'nitrogen'].includes(name)) {
+          setFormCustomFertilizer((prev) => ({ ...prev, ...changes }));
+        }
 
-    if (name === 'fertilizerId') {
-      // eslint-disable-next-line no-param-reassign
-      changes.name = filteredFertilizersOptions.find((f) => f.id === value)!.name;
-    }
+        if (name === 'fertilizerTypeId') {
+          setFilteredFertilizerOptions(
+            fertilizerOptions.filter(
+              (ele) =>
+                ele.dryliquid ===
+                fertilizerTypes.find((fertType) => fertType.id === value)?.dryliquid,
+            ),
+          );
 
-    setFormState((prev) => ({ ...prev, ...changes }));
-  };
+          if (value === 2) {
+            // For custom dry fertilizers
+            setFormCustomFertilizer((prev) => ({ ...prev, dryliquid: 'dry' }));
+          } else if (value === 4) {
+            // For custom liquid fertilizers
+            setFormCustomFertilizer((prev) => ({ ...prev, dryliquid: 'liquid' }));
+          } else {
+            // Reset for other values
+            setFormCustomFertilizer(EMPTY_CUSTOM_FERTILIZER);
+          }
+
+          // Reset other values on changes
+          changes = { ...structuredClone(EMPTY_FERTILIZER_FORM_DATA), fertilizerTypeId: value };
```

Comment:
You don't need to clone an object you're deconstructing with `...`, this operation makes a new object.

Reply:
Removed!